python/cugraph/structure/graph_wrapper.pyx

```diff
@@ -74,6 +74,16 @@ def add_edge_list(graph_ptr, source_col, dest_col, value_col=None):
     cdef uintptr_t graph = graph_ptr
     cdef gdf_graph * g = <gdf_graph*> graph
 
+    # Checks in python
```

Comment:
Migrating checks from cpp to python:
-> Checks if src col and dest col lengths are same
-> Typecasts dtype of src and dest to int32 if not already int32
-> Typecasts dtype of value(weights) to float32 if not already float32 or float64
-> Null_count checking to be added?

Reply:
-> Typecasts dtype of value(weights) to float32 if not already float32 or float64
We should allow othertype, the cast should happen at the time an algo requiring fp32/64 is called

Reply:
-> Typecasts dtype of src and dest to int32 if not already int32
we may want to leverage renumbering for that

Reply:
Overall, I'd recommend to first update the python graph structure before adding more automation at the python level

Reply:
The edge weights reside in the form of a cpp gdf_column in the graph structure. If we save the graph with user-input datatypes into our gdf_G structure and only typecast it when algo requiring fp32/64 is called, since the api takes in only pointer to gdf_graph, typecasting could either be done by manipulating the graph edge weights itself (which is, changing the user-input in saved graph, would require a cpp function to be called) or sending in edge weights as a separate column (not ideal)